### DISCUSSION TITLE
在扩展识别模型的词典时是否可以继承之前的fc weight？

### DISCUSSION THREAD

**Discussion by flow3rdown** [PaddlePaddle/PaddleOCR#14152] [Q&A]
扩展识别模型的词典大小后，FC weight的维度发生了变化，但是之前的词表是全部保留的，这种情况下是否可以保留之前的FC weight，新增扩展词的weight并与之前的FC weight进行合并？这样可行吗？

**Comment by flow3rdown** [comment]
有人能解答一下吗？

**Reply by UserWangZz** [reply]
感觉不太行，不如试试单独训练FC层？

**Reply by flow3rdown** [reply]
> 感觉不太行，不如试试单独训练FC层？

语言模型扩展词向量一般都只是新建一些层，与之前的词向量合并。我不太清楚OCR是否可以这样做，直观上感觉保留之前的FC weight要比重新初始化会更容易训练一点，或者对数据的要求会更少一点？但我没具体实验尝试过所以想问下您是否有此方面的经验？

**Reply by UserWangZz** [reply]
这个不太一样吧，最后一层的FC类似于做一个分类操作，单独的直接合并FC层感觉不是一个有效的操作，这个我也没有实验过，不敢保证
